Does it make sense to mention "windows" in the commit message?

BTW: Your points 2 and 3 are the same for all single-configuration generators (GNU/NMake/Ninja Makefiles) where the executable is built directly in the root dir. They are only really different on multi-configuration generators (Visual Studio and Xcode). They allow to specify the configuration (Release/Debug) at build-time and need separate paths where they put the executables.

Fails to compile on Linux:
```
xbmc/Application.cpp: In member function ‘bool CApplication::InitDirectoriesLinux()’:
xbmc/Application.cpp:883:47: error: no matching function for call to ‘CUtil::GetHomePath(std::__cxx11::string&, const char*&)’
   CUtil::GetHomePath(appBinPath, envAppBinHome);
                                               ^
In file included from xbmc/Application.cpp:32:0:
xbmc/Util.h:63:22: note: candidate: static std::__cxx11::string CUtil::GetHomePath(std::__cxx11::string)
   static std::string GetHomePath(std::string strTarget = "KODI_HOME"); // default target is "KODI_HOME"
                      ^
xbmc/Util.h:63:22: note:   candidate expects 1 argument, 2 provided
```

--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -872,7 +872,7 @@ bool CApplication::InitDirectoriesLinux()
   else
     userHome = "/root";
 
-  std::string appBinPath, appPath;
+  std::string appPath;
   std::string appName = CCompileInfo::GetAppName();
   std::string dotLowerAppName = "." + appName;
   StringUtils::ToLower(dotLowerAppName);
@@ -880,7 +880,7 @@ bool CApplication::InitDirectoriesLinux()
   const char* envAppBinHome = "KODI_BIN_HOME";
   const char* envAppTemp = "KODI_TEMP";
 
-  CUtil::GetHomePath(appBinPath, envAppBinHome);
+  auto appBinPath = CUtil::GetHomePath(envAppBinHome);
   if (getenv(envAppHome))
     appPath = getenv(envAppHome);
   else
@@ -961,8 +961,7 @@ bool CApplication::InitDirectoriesOSX()
   else
     userHome = "/root";
 
-  std::string appPath;
-  CUtil::GetHomePath(appPath);
+  std::string appPath = CUtil::GetHomePath();
   setenv("KODI_HOME", appPath.c_str(), 0);
 
 #if defined(TARGET_DARWIN_IOS)

--- a/xbmc/Util.cpp
+++ b/xbmc/Util.cpp
@@ -239,7 +239,7 @@ std::string GetHomePath(const std::string& strTarget, std::string strPath)
 {
   if (strPath.empty())
   {
-    auto strHomePath = CUtils::ResolveExecutablePath();
+    auto strHomePath = CUtil::ResolveExecutablePath();
     size_t last_sep = strHomePath.find_last_of(PATH_SEPARATOR_CHAR);
     if (last_sep != std::string::npos)
       strPath = strHomePath.substr(0, last_sep);

